src/parser/utils.ts

```diff
@@ -1,31 +1,54 @@
+const map = new WeakMap<object, string>();
+
+/**
+ * Retrieves the prototype.toString() of a value.
+ * If the value is an object, it will cache the result in a WeakMap for future use.
+ */
+function getPrototypeString(value: unknown): string {
+  let str = map.get(value as object);
+
+  if (!str) {
+    str = Object.prototype.toString.call(value);
+    if (value !== null && typeof value === 'object') {
+      map.set(value, str);
+    }
+  }
+  return str;
+}
+
 export function isAnyArrayBuffer(value: unknown): value is ArrayBuffer {
-  return ['[object ArrayBuffer]', '[object SharedArrayBuffer]'].includes(
-    Object.prototype.toString.call(value)
-  );
+  const type = getPrototypeString(value);
+  return ['[object ArrayBuffer]', '[object SharedArrayBuffer]'].includes(type);
```

Comment:
Since we're trying to get even the slightest of gains, what do you think about changing all the strings here in this module to be constants? This way we're not instantiating a new string or array of strings for each call but rather referencing the constant? So for example in this case:

```ts
const ARRAY_BUFFER = ['[object ArrayBuffer]', '[object SharedArrayBuffer]'].freeze;

// Then in the isAnyArrayBuffer function
return ARRAY_BUFFER.includes(type);
```

Thoughts?

Reply:
I like these ideas. We could do something like this

```ts
const TYPES = {
  ArrayBuffer: '[object ArrayBuffer]',
  SharedArrayBuffer: '[object SharedArrayBuffer]',
  Uint8Array: '[object Uint8Array]',
  ...
}

// Then in the isAnyArrayBuffer function 
return type === TYPES.ArrayBuffer || type === TYPES.SharedArrayBuffer
```

Not sure if there's going to be any measurable performance improvement here but in theory I think it's more efficient

Reply:
Yeah I think in terms of performance it will be minimal but would at least decrease string instantiation and potential load on GC. The suggestion you propose I think is nice as well.